contracts/TwabRewards.sol

```diff
@@ -255,10 +265,14 @@ contract TwabRewards is ITwabRewards {
         @param _promotion Promotion to check
     */
     function _requirePromotionActive(Promotion memory _promotion) internal view {
-        uint256 _promotionEndTimestamp = _promotion.startTimestamp +
-            (_promotion.epochDuration * _promotion.numberOfEpochs);
-
-        require(_promotionEndTimestamp >= block.timestamp, "TwabRewards/promotion-inactive");
+        unchecked {
```

Comment:
This should also require that the promotion has started, no?

Reply:
Yes if we only want to allow to end or extend a promotion between the start timestamp and the end timestamp of the promotion.
Here, we consider that a promotion is active until it has ended. It would add complexity if we decide to check that block.timestamp is between start and end timestamp. 
Also not sure it would make much of a difference, at least it allows a promotion creator to retrieve funds if the promotion has not started yet.

Reply:
I see what you're saying; it allows cancellation and extension prior to the start of the promotion.  Ok